org.eclipse.scout.rt.server/src/main/java/org/eclipse/scout/rt/server/servicetunnel/ServiceTunnelService.java

```diff
@@ -63,38 +65,61 @@ public void init() {
+  }
+
+  /**
+   * Default handling, writes {@link ServiceTunnelResponse} directly to given {@link OutputStream}.
+   */
+  protected Response handleResponse(ServiceTunnelResponse response) {
+    StreamingOutput stream = out -> serializeServiceResponse(out, response);
+    return Response.ok(stream).build();
   }
 
-  // incoming request
-  public void incomingRequest(InputStream in, OutputStream out) {
+  /**
+   * Handles incoming request by given {@link InputStream} and delegates response handling to given {@link ServiceTunnelResponseHandler}.
+   */
+  protected Response handleIncomingRequest(InputStream in, ServiceTunnelResponseHandler responseConsumer) {
     if (Subject.current() == null) {
       throw new ForbiddenException();
     }
 
-    m_serverRunContextProducer.get()
-        .getInnerRunContextProducer()
-        .produce(IHttpServletRoundtrip.CURRENT_HTTP_SERVLET_REQUEST.get(), IHttpServletRoundtrip.CURRENT_HTTP_SERVLET_RESPONSE.get())
-        .withProperties(enableSignature(IHttpServletRoundtrip.CURRENT_HTTP_SERVLET_REQUEST.get()) ? Map.of(ServiceTunnelOptions.ID_SIGNATURE_PROP, true) : Map.of())
-        .run(() -> {
-          ServiceTunnelRequest serviceRequest = deserializeServiceRequest(in);
-          ServiceTunnelResponse serviceResponse = evaluate(serviceRequest);
-
-          // Clear the current thread's interruption status before writing the response to the output stream.
-          // Otherwise, the stream gets silently corrupted, which triggers  a repetition of the current request by Java connection mechanism.
-          IRestorer interruption = ThreadInterruption.clear();
-          try {
-            serializeServiceResponse(out, serviceResponse);
-          }
-          finally {
-            interruption.restore();
-          }
-        }, ServiceTunnelExceptionTranslator.class);
+     // Add signature property to current run context if requested by current request. Property is required for correct deserialization of request and serialization of response.
+     // Serialize of response is performed deferred by StreamingOutput lambda response provided by #handleDefault which runs within the same run context.
+    if (enableSignature(IHttpServletRoundtrip.CURRENT_HTTP_SERVLET_REQUEST.get())) {
+      RunContext.CURRENT.get().getPropertyMap().put(ServiceTunnelOptions.ID_SIGNATURE_PROP, true);
+    }
+
+    // FIXME PBZ,MOT: Check if this m_serverRunContextProducer is necessary -> org.eclipse.scout.rt.server.context.ServerRunContextFilter already provided same/similar RunContext
+//    return m_serverRunContextProducer.get()
+//        .getInnerRunContextProducer()
+//        .produce(IHttpServletRoundtrip.CURRENT_HTTP_SERVLET_REQUEST.get(), IHttpServletRoundtrip.CURRENT_HTTP_SERVLET_RESPONSE.get())
+//        //.withProperties(enableSignature(IHttpServletRoundtrip.CURRENT_HTTP_SERVLET_REQUEST.get()) ? Map.of(ServiceTunnelOptions.ID_SIGNATURE_PROP, true) : Map.of())
+//        .call(() -> {
```

Comment:
We probably do not need the additional `RunContext` here, previous purpose was I believe to add the ìd signature run context property (which we now add directly to the property map, see https://github.com/eclipse-scout/scout.rt/pull/1806/files#diff-76a72b3ca6d38aa5eca2d2ee56f4066ac28f3646599378e9d1f7bc4135e21635R96).